Implement quiz navbar and handle end of quiz when questions run out

fungjj92 opened the pull request:
## Overview

This PR implements the navbar that shows the user their progress throughout the quiz. It also implements the quiz successfully saving quiz result information and closing the quiz when the questions are finished. Score to be further worked on in #72 

Connects #44 

### Demo

<img width="1417" alt="Screen Shot 2019-06-06 at 4 54 46 PM" src="https://user-images.githubusercontent.com/10568752/59065924-db1d5b80-887b-11e9-846a-ae9f7ad61682.png">
<img width="879" alt="Screen Shot 2019-06-06 at 4 55 02 PM" src="https://user-images.githubusercontent.com/10568752/59065925-db1d5b80-887b-11e9-814e-4f6159bcca66.png">

It is a little tough to see, but will be changed in the future of course, a green underline in the navbar denoting which question you're on.

### Notes

Follow up card to flesh out the fish icons in the quiz navbar, pending budget renewal in July.

I included some other clean up around the app, like refactoring fish names, cleaning up the guess page, and fitting Meet the Fish to the official screen resolution of 1920x1080px !

## Testing Instructions

Play the quiz! :) See that there's no console errors, screen sizing fits a resolution of 1920x1080, and the navbar tracks the quiz question. For now the would-be score page is just a "try again" button. Upon navigating away from the quiz tab, and back, the quiz should be reset.

maurizi committed:
Add questions state and navbar

fungjj92 committed:
Allow fontStyle prop on custom text components

Prepare these proptypes to style the in/correct guess component

Refs #44

fungjj92 committed:
Update quiz guess component to match wireframes

Refs #44

fungjj92 committed:
Close quiz view upon quiz completion

Refs #44

fungjj92 committed:
Close quiz upon completion but persist results to redux

Refs #44

fungjj92 committed:
Resize meet the fish reel to fit ELO resolution

Refs #73

fungjj92 committed:
Increase quiz questions back to 5

Refactor said value into a constant

Refs #44

fungjj92 committed:
Refactor fish naming into reusable component

We name the fish identically in name places, yet wrote out different styling each time. Instead, refactor all named instanced into a reusable component for uniformity and ease.

caseycesari commented on src/app/package.json:
```diff
@@ -48,6 +48,7 @@
     "jest-resolve": "23.6.0",
     "jest-watch-typeahead": "^0.2.1",
     "leaflet": "~1.3.1",
+    "lodash-es": "^4.17.11",
```
We should use the regular `lodash` library and do imports like `import foo from 'lodash/foo'` to support tree-shaking. See this helpful blog post for more info: https://www.azavea.com/blog/2019/03/07/lessons-on-tree-shaking-lodash/

fungjj92 replied:
How apropro 👌 Forgot you did that research. Other research I read out there was more borderline on lodash-es v lodash. Will make this change 👍

caseycesari commented on src/app/src/components/QuizQuestion.js:
```diff
@@ -58,7 +58,9 @@ class QuizQuestion extends React.Component {
 
         const showHint = guessed.length > 0 || usedHint;
         // TODO: Remove the default value here when we get hint text
```
I think this comment can go now, right?

fungjj92 replied:
Hm, it could be modified but not go. We're still providing a dummy hint as a fallback that we'll want to erase when hints exist.

caseycesari reviewed (approved):
Working well! Nice job.

fungjj92 committed:
Use lodash v4 instead of lodash-es

Lodash v4 + function imports + webpack does tree-shaking out of the box. See https://www.azavea.com/blog/2019/03/07/lessons-on-tree-shaking-lodash/

Refs #44

fungjj92 commented:
Thank you! ^_^ & @maurizi
